vnext/Microsoft.ReactNative/TurboModulesProvider.cpp

```diff
@@ -357,6 +371,7 @@ class TurboModuleImpl : public facebook::react::TurboModule {
   winrt::com_ptr<TurboModuleBuilder> m_moduleBuilder;
   IInspectable m_providedModule;
   std::shared_ptr<implementation::HostObjectWrapper> m_hostObjectWrapper;
+  std::shared_ptr<facebook::jsi::HostObject> m_nonAbiHostObject;
```

Comment:
Up for debate, but we could presumably just have one shared_ptr here, since implementation::HostObjectWrapper derives from facebook::jsi::HostObject